eth/gasprice/gasprice.go

```diff
@@ -17,212 +17,152 @@
 package gasprice
 
 import (
+	"context"
 	"math/big"
-	"math/rand"
+	"sort"
 	"sync"
 
-	"github.com/ethereum/go-ethereum/core"
-	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/ethdb"
-	"github.com/ethereum/go-ethereum/event"
-	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/rpc"
 )
 
-const (
-	gpoProcessPastBlocks = 100
-
-	// for testing
-	gpoDefaultBaseCorrectionFactor = 110
-	gpoDefaultMinGasPrice          = 10000000000000
-)
-
-type blockPriceInfo struct {
-	baseGasPrice *big.Int
-}
-
 type GpoParams struct {
-	GpoMinGasPrice          *big.Int
-	GpoMaxGasPrice          *big.Int
-	GpoFullBlockRatio       int
-	GpobaseStepDown         int
-	GpobaseStepUp           int
-	GpobaseCorrectionFactor int
+	GpoBlocks     int
+	GpoPercentile int
+	GpoDefault    *big.Int
```

Comment:
There's no need for these fields to have the `Gpo` prefix. Every field is a configuration parameter of the gas price oracle. No need to remind everywhere.